Readme.md

```diff
@@ -1,15 +1,86 @@
-### Lattice Surgery Compiler Serverless Deployment
+# Lattice Surgery Compiler's API Service
 
-Deploys a lambda containing Python code and sets up an API gateway to call it. Soon this code will be the compiler.
+[![Unitary Fund](https://img.shields.io/badge/Supported%20By-UNITARY%20FUND-brightgreen.svg?style=for-the-badge)](http://unitary.fund)
 
-Requires the AWS cli tools with credentials set up and Python 3.8.
+Sets up a [serverless stack](https://aws.amazon.com/serverless/) to deploy the [Lattice Surgery Error Correcting Compiler](https://github.com/latticesurgery-com/lattice-surgery-compiler) as service at `api.latticesurgery.com`.
 
-To deploy the first time, run
+## How to use
+
+Send post requests to `https://api.latticesurgery.com/compile`. **Example** with cURL:
 ```
-$ ./1-create-bucket.sh
-$ ./2-build-layer.sh
-$ ./3-deploy.sh
+curl -X POST https://api.latticesurgery.com/compile -d"{\"circuit_source\":\"str\",\"circuit\":\"OPENQASM 2.0;\\ninclude \\\"qelib1.inc\\\";\\n\\nqreg q[2];\\n\\nh q[0];\\ncx q[0],q[1];\\nh q[0];\\nt q[1];\\ns q[0];\\nx q[0];\",\"apply_litinski_transform\":true}"
 ```
-After the first time running 2 and 3 is enough.
 
-The other scripts should be self explainatory.
+The input and output are specified in the [compiler's pipeline](https://github.com/latticesurgery-com/lattice-surgery-compiler/blob/a7208d1efe9229fd4b17ffb19ea1fcfa71c18b3f/webapi/json_pipeline.py#L33).
+
+If you are developing [lattice-surgery-compiler](https://github.com/latticesurgery-com/lattice-surgery-compiler) on your local machine, ignore this repo and use [the local development server](https://github.com/latticesurgery-com/lattice-surgery-compiler/blob/master/webapi/local_compiler_api_server.py) instead.
+
+## Set Up
+
+### Dependecies
+ * Python 3.9
+ * Python Pip
+ * Docker with daemon running
+ * AWS CLI >= 1.19
+ * Git version with support for submodules
+ 
+#### Qiskit's build Dependecies 
+Building Qiskit (i.e. running pip install) seems to have a slew of dependecies. Fortunately these don't get to the deployed code, but they can make setting up the environment on the deployment host tedious. Some things I noticed to be necessary are:
+ * CMAKE
+ * gcc
+ * g++
+ * Bison + Yacc
+ 
+And probably more... I would expect a machine with basic build tools to have most of these (my laptop did), however be prepared to have to install some software, especially if deploying from a minimal installation (like on a fresh EC2 instance - I tried). If qiskit fails to build, then it's very likeley that because of missing some build dependncy. Try pasting the error message into Google.
```

Comment:
```suggestion
And probably more... I would expect a machine with basic build tools to have most of these (my laptop did), however be prepared to have to install some software, especially if deploying from a minimal installation (like on a fresh EC2 instance - I tried). If Qiskit fails to build, then it's very likely that because of missing some build dependencies. Try pasting the error message into Google.
```